### ISSUE TITLE
LineLength false positive - platform encoding

### ISSUE THREAD

**Issue by george-007** [jshiell/checkstyle-idea#295]
Intelij idea 2016.3.4, checkstyle plugin 5.1.4, global + project encoding UTF-8, Windows (platform encoding cp-1250).

Checkstyle plugin probably uses platform encoding when working with files and that gives false positives for LineLength when javadoc contains local characters.

checkstyle.xml:
```
<module name="LineLength">
    <property name="max" value="5"/>
</module>
```

No warning (linelength=5):
``
//345
``

Warning (linelength=5):
``
//34č
``

![encoding1](https://cloud.githubusercontent.com/assets/1450087/22647817/d780e902-ec73-11e6-8a25-b925b8f19227.png)
![encoding2](https://cloud.githubusercontent.com/assets/1450087/22647818/d781be72-ec73-11e6-953a-fd1b5cfd041c.png)

mvn ``checkstyle:check`` gives no warning.

**Comment by jshiell** [comment]
Thanks for the report. I miss a lot of these issues, as I develop on macOS where the native encoding is UTF-8. I'll try have a look this weekend.

**Comment by jshiell** [comment]
No progress on this so far. I'm going mostly via inspection at present, as it's hard to reproduce on a platform where everything lines up as UTF-8.

We have two paths - if the file is modified, we write a temporary file to disk and scan that, otherwise we give the original filename to Checkstyle. **If you could let me know if the problem occurs when the file is unsaved, saved or both it would be most helpful.**

When creating the temporary file we currently force the output to UTF-8; I'm adding a fix to make this more correct, by asking IDEA what charset the file should be, but this doesn't sound like it'd affect this particular issue. In the latter case we don't touch the file content at all, which makes it more interesting.

Which leads me to my second question - **does your rules file set the character encoding?** From memory, it's the `charset` property of the `Checker`. Thanks!

**Comment by george-007** [comment]
I'm not sure what you mean exactly - ``Check current file`` on a not modified file gives warning. When editing the file I got new warnings too. 

Regarding the second question - my rules file doesn't set the character encoding and adding ``charset`` property to the ``checker`` module indeed fixes the issue for me:
``` 
<module name="Checker">
  <property name="charset" value="UTF-8"/>
```
According to the checkstyle doc the default value of charset is	system property ``file.encoding``.

Note - the maven checkstyle plugin sets the default encoding from ``${project.build.sourceEncoding}`` which is ''UTF-8'' in my case and that is why ``mvn checkstyle:check`` gives no warnings for me.

**Comment by jshiell** [comment]
Ah, that'd explain it then - we pass unmodified files directly to Checkstyle, and therefore CS is responsible for the char encoding. The Maven results were confusing me, but since it takes it from source encoding then that explains it.

I'm going to close this, as we should rely on the Checkstyle behaviour (given that an IDEA project may have multiple encodings across various files). Thanks for your help.
